python/lsst/ts/salobj/sal_info.py

```diff
@@ -781,7 +785,8 @@ def _blocking_read(
         conditions = self._waitset.wait(self._wait_timeout)
         if not self.isopen:
             # shutting down; clean everything up
-            return (None, [])
+            return None
```

Comment:
Please return an empty list, to simplify the API.

Reply:
Please update the doc string to something along these lines:
```
"""Read DDS data

Returns
-------
read_data
    Read data, as a list of tuples, each with two values:

    * reader : `ReadTopic`
        The read topic.
    * data_list: `list` [`BaseMsgType`]
        List of one or more messages.
"""
```
The most important changes are:
* reader cannot be None, so simplify that description.
* data_list cannot be empty (if you update the code below as requested).
* Move the return description to a Returns section, fixing an existing issue.